### ISSUE TITLE
Filter scanlators by priority

### ISSUE THREAD

**Issue by scb261** [tachiyomiorg/tachiyomi#8804]
### Describe your suggested feature

TLDR: User can set up which scanlators he prefers over others. If first priority is not available, it would fallback to the second. If not, than third. And so on.

For example there is a manga with following chapters:
- ch. 1, group A
- ch. 2, group A
- ch. 2, group B
- ch. 2, group C
- ch. 3, group B

Current implementation (#8803) allows to filter out only whole groups from manga, so user would end up without chapter 1 or without chapter 3.

With priorities the user can select that he prefers groups in specific order, for example A > B > C. Then the filter would show each chapter only once like this:
- ch. 1, group A
- ch. 2, group A
- ch. 3, group B

### Other details

Related: #8803, #969

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.14.2](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by WalterStager** [comment]
Looking at implementing this based on the filter implementation by [AntsyLich](https://github.com/AntsyLich) (https://github.com/tachiyomiorg/tachiyomi/pull/8803)
https://github.com/WalterStager/antsyomi/tree/chore/scanlatorFilter

Looking at other parts of the app I don't think there are arrange-able lists apart from the "Edit Categories" list but the scanlator list would be used more often so arrows for moving items single steps could be difficult. Please mention if there is some other arrange-able list functionality already implemented.

We will see though, I am not an experience android developer.

**Comment by AntsyLich** [comment]
There are no other rearrangeable list in Tachiyomi atm. Also do give me co-author if you base it off my implementation

**Comment by avstoyanov** [comment]
This would be really nice to see. It would solve the issue with duplicates and make sources with multiple scanlators actually usable. As it stands the only sane option is to find the best scanlator and migrate the series to use them as a source, which is hardly anywhere as reliable as having multiple possibilities and backups.
